bytecomp/matching.ml

```diff
@@ -1160,7 +1196,20 @@ let rec split_or argo (cls : Half_simple.clause list) args def =
   in
   do_split [] [] [] cls
 
-and split_no_or cls args def k =
+and split_no_or (cls : Simple.clause list) args def k =
+  (* We split the remaining clauses in as few pms as possible while maintaining
+     the property stated earlier (cf. comment at the top of the file), i.e. for
```

Comment:
I would name the comments and refer to the comment by name. (This is more robust to moving content to other files, etc.; and it is how GHC does it.)

Reply:
I don't know how you'd do that, using ocamldoc's reference syntax?
Why don't you push that change on this branch?

Reply:
I had a low-tech solution in mind, using custom markup syntax that is not recognized by ocamldoc: eg.

```
Note {pattern compilation overview}: ...
```

```
... see the note {compilation overview} ...
```

I can push this later, if you haven't done it.

Reply:
I pushed something along these lines.